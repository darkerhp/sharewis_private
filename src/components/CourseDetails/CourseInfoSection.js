import React from 'react';
import ReactNative from 'react-native';
import I18n from 'react-native-i18n';
import { _ } from 'lodash';

import BaseStyles from '../../baseStyles';
import NextLectureArea from './NextLectureArea';
import Duration from '../Duration';
import Progress from './Progress';

const { PropTypes } = React;
const { Platform, StyleSheet, Text, View } = ReactNative;

const styles = StyleSheet.create({
  container: { padding: 10 },
  lectureAreaContainer: { flex: 3 },
  courseTitleWrapper: { flex: 1 },
  courseTitleText: {
    color: BaseStyles.textColor,
    fontSize: 16,
    fontWeight: '900',
  },
  totalDurationWrapper: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
  },
  totalDuration: {
    ...Platform.select({
      android: {
        width: 130,  // Or text will be trimmed in english
      },
    }),
    textAlign: 'center',
    color: BaseStyles.textColor,
    fontSize: 10,
    fontWeight: 'bold',
    padding: 5,
    backgroundColor: '#F2F2F2',
  },
});


const CourseInfoSection = ({
  // values
  completeLectureCount,
  containerStyle,
  courseTitle,
  nextLecture,
  totalLectureCount,
  totalDuration,
  // actions
  handlePressNextLecture,
}) =>
  <View style={[styles.container, containerStyle]}>
    <View style={styles.courseTitleWrapper}>
      <Text style={styles.courseTitleText}>{courseTitle}</Text>
    </View>

    {!_.isEmpty(nextLecture) &&
      <NextLectureArea
        nextLecture={nextLecture}
        handlePressNextLecture={handlePressNextLecture}
        hidden
        containerStyleId={styles.lectureAreaContainer}
      />
    }

    <Progress
      completeLectureCount={completeLectureCount}
      totalLectureCount={totalLectureCount}
    />

    <Duration
      estimatedTime={totalDuration}
      format={I18n.t('totalDurationFormat')}
      containerStyleId={styles.totalDurationWrapper}
      durationStyleId={styles.totalDuration}
    />
  </View>;


CourseInfoSection.propTypes = {
  courseTitle: PropTypes.string.isRequired,
  totalLectureCount: PropTypes.number.isRequired,
  completeLectureCount: PropTypes.number.isRequired,
  totalDuration: PropTypes.number.isRequired,
  nextLecture: PropTypes.shape({}).isRequired,
  handlePressNextLecture: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  containerStyle: PropTypes.object.isRequired,
};

export default CourseInfoSection;
