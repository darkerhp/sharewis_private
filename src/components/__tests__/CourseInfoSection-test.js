/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import CourseInfoSection from '../CourseDetails/CourseInfoSection';

jest.mock('bugsnag-react-native', () => 'Bugsnag');
jest.mock('react-native-i18n', () => ({ t: () => '' }));
jest.mock('../Duration', () => 'Duration');
jest.mock('../CourseDetails/NextLectureArea', () => 'NextLectureArea');
jest.mock('../CourseDetails/Progress', () => 'Progress');

describe('CourseInfoSection', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <CourseInfoSection
        completeLectureCount={0}
        containerStyle={{}}
        courseTitle={'Title'}
        isCompleted={false}
        nextLecture={{}}
        totalLectureCount={10}
        totalDuration={100}
        handlePressNextLecture={() => null}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
