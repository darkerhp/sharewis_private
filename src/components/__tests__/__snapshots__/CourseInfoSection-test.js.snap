exports[`CourseInfoSection renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "padding": 10
      },
      Object {}
    ]
  }>
  <View
    style={
      Object {
        "flex": 1
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "900"
        }
      }>
      Title
    </Text>
  </View>
  <NextLectureArea
    containerStyleId={
      Object {
        "flex": 3
      }
    }
    handlePressNextLecture={[Function handlePressNextLecture]}
    hidden={true}
    nextLecture={Object {}} />
  <Progress
    completeLectureCount={0}
    totalLectureCount={10} />
  <Duration
    containerStyleId={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "flex-end"
      }
    }
    durationStyleId={
      Object {
        "backgroundColor": "#F2F2F2",
        "color": "black",
        "fontSize": 10,
        "padding": 5,
        "textAlign": "center",
        "width": 130
      }
    }
    estimatedTime={100}
    format="[missing \"en.totalDurationFormat\" translation]" />
</View>
`;
